File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.java
class Solution {
    public int[] findMissingAndRepeatedValues(int[][] grid) {
        int n=grid.length;
        int N=n*n;
        int sn=(N*(N+1))/2;
        int s2n=(N*(N+1)*(2*N+1))/6;
        int sum1=0;
        int sum2=0;
        for(int i=0;i<n;i++)
        {
            for(int j=0;j<n;j++)
            {
                sum1+=grid[i][j];
                sum2+=grid[i][j]*grid[i][j];
            }
        }
        int val1=sn-sum1;
        int val2=s2n-sum2;
        val2=val2/val1;
        int x=(val1+val2)/2;
        int y=x-val1;
        return new int[]{y,x};
    }
}

File: 2965-find-missing-and-repeated-values/2965-find-missing-and-repeated-values.py
class Solution:
    def findMissingAndRepeatedValues(self, grid: List[List[int]]) -> List[int]:
        n = len(grid)
        N = n * n
        sn = (N * (N + 1)) // 2
        s2n = (N * (N + 1) * (2 * N + 1)) // 6

        sum1 = 0
        sum2 = 0

        for i in range(n):
            for j in range(n):
                sum1 += grid[i][j]
                sum2 += grid[i][j] ** 2

        val1 = sn - sum1
        val2 = (s2n - sum2) // val1
        x = (val1 + val2) // 2
        y = x - val1

        return [int(y), int(x)]
